lab_4_summarization_textrank/main.py

```diff
@@ -84,31 +106,59 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
         """
         Constructs all the necessary attributes
         """
-        pass
+        check_type(stop_words, tuple, str, can_be_empty=True)
+        check_type(punctuation, tuple, str)
+        super().__init__(stop_words, punctuation)
 
     def _split_by_sentence(self, text: str) -> tuple[Sentence, ...]:
         """
         Splits the provided text by sentence
         :param text: the raw text
         :return: a sequence of sentences
         """
-        pass
+        # sentences = re.split('[.?!]\s(?=[A-ZА-ЯЁ])', text)
+        check_type(text, str)
+        clean_text = (' '.join(text.split())).replace('!', '.').replace('?', '.')
+        sentences = []
+        sentence_start_idx = 0
+        sentence_counter = 0
+
+        for idx in range(len(clean_text) - 2):
```

Comment:
why `-2`?

Reply:
cause we look at 3 characters simultaneously: one of the certain index and two following ones, to see if this 3-character sequence consists of a full stop, whitespace and capital respectively

Reply:
now I see, thanks! good